### PR TITLE
#2511: extended reth-rpc example with custom rpc ext

### PR DESCRIPTION
**Pull request by loocapro**
ref #2511

- Refactored the structure of the example as a single cargo project
- Used `with_noop_pool` and `with_noop_network` methods from `RpcModuleBuilder` instead of their Defaults
- Added `MyRpcExt` to showcase how to add custom methods on this example

### CODE REVIEW

File: examples/rpc-db/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+reth-blockchain-tree.workspace = true
+reth-db.workspace = true
+reth-network-api.workspace = true
+reth-primitives.workspace = true
+reth-provider.workspace = true
+reth-revm.workspace = true
+reth-rpc = { version = "0.1.0-alpha.8", path = "../../crates/rpc/rpc" }
+reth-rpc-builder.workspace = true
+reth-tasks.workspace = true
+reth-transaction-pool.workspace = true
```
Comment (mattsse) [inline examples/rpc-db/Cargo.toml:22]: I don't think we need all of this most of this is reexported from reth itself

https://github.com/paradigmxyz/reth/blob/685148117aab83d4b38f2e05adb8c6c96943f7db/examples/additional-rpc-namespace-in-cli/Cargo.toml#L8-L14

**Review by mattsse — CHANGES_REQUESTED** [review]

File: bin/reth/src/utils.rs [hunk 2]
```diff
@@ -26,6 +26,11 @@ use std::{
 };
 use tracing::info;
 
+/// Exposing the `reth_db` crate
+pub mod utils {
```
Comment (lucaprov1989) [inline bin/reth/src/utils.rs:30]: I was not quite sure where to put this, but seemed like the function needed to be exposed.

File: examples/rpc-db/Cargo.toml (hunk 1 above)
Comment (lucaprov1989) [reply examples/rpc-db/Cargo.toml]: Had to export a bit of stuff, but should be good now!

File: bin/reth/src/utils.rs (hunk 2 above)
Comment (mattsse) [reply bin/reth/src/utils.rs:30]: agree, but I feel like this should be called `db`

File: bin/reth/src/lib.rs [hunk 3]
```diff
@@ -46,6 +46,25 @@ pub mod providers {
...
+}
+/// Re-exported from `reth_blockchain_tree`.
+pub mod blockchain_tree {
+    pub use reth_blockchain_tree::*;
+}
+
+/// Re-exported from `reth_revm`.
+pub mod revm {
+    pub use reth_revm::*;
+}
```
Comment (mattsse) [inline bin/reth/src/lib.rs:66]: all of these are reasonable

File: bin/reth/src/lib.rs [hunk 4]
```diff
@@ -79,6 +98,10 @@ pub mod rpc {
     pub mod api {
         pub use reth_rpc_api::*;
     }
+    /// Re-exported from `reth_rpc::eth`.
+    pub mod eth {
+        pub use reth_rpc::eth::*;
+    }
```
Comment (mattsse) [inline bin/reth/src/lib.rs:104]: I think this is fine since eth contains a bunch of useful code.

**Review by mattsse — CHANGES_REQUESTED** [review]
last naming nit

**Comment by lucaprov1989** [comment]
> last naming nit

Done thanks!

**Review by mattsse — APPROVED** [review]

**Pull request merged**